specs/pluggable_alert_delivery.adoc

```diff
@@ -0,0 +1,372 @@
+        alert has been notified. When alerting module is started, it scans all
+        alerts in '/alerts' directory and sends out notifications for those
+        that are marked significance 'HIGH' but are not set with
+        'notified':True
+  ** An alert will also have a field to ack it. Acking an alert can be done by
+     tendrl-user(lets say user is aware of a problem and has taken some action
+     on it and doesn't want to be shown the alert until the issue is completely
+     rectified. In such case user can ack such an alert using an api that
+     tendrl-api exposes and such an api will set 'acked': True and
+     'acked-by': <user_name> to indicate user <user-name> acked the alert and
+     doesn't want to see it in the list of alerts) or occurence of negation of
+     existing alert(ex: Lets say cpu utilization of a node breached critical
+     threshold and a corresponding alert is raised by collectd and the alert is
+     stored in etcd by the node-agent after a while when the cpu utilization is
+     back normal, previous alert corresponding to its threshold breach needs
+     to be acked as no more relevant. This is achieved by setting 'acked' to True
+     and 'ackedby' to tendrl which means tendrl has acked this alert as no more
+     relevant)
+  ** In addition to the above, the alert contains details such as the id of the
+     node on which the alert was detected, the time stamp at which the alert
+     was observed, resource for which the alert was generated, the type of
+     alert, severity of alert etc... as described in section
+     "Data model impact".
+* The Tendrl/alerting module watches for new alerts in etcd's /alerts directory
+  and sends out notifications to configured destinations via configured means
+  ex: mails or sms or snmp traps.
+  The different notification medium will be hooked into tendrl in a pluggable
+  manner as follows:
+  ** Handlers will be defined in alerting module for dispatching observed
+     alerts to configured destinations through respective means.
+     ex: mail handler for sending out mail and snmp handler for snmp-trap
+    *** A new handler can be hooked into the tendrl alerting system by adding
+        it to a designated directory and with a name prefixed with notification
+        medium name. ex: email_handler for mail notification in the directory
+        alerting/tendrl/alerting/notification/
+    *** Whenever a new handler is added, the first thing the handler is
+        supposed to do is add itself as a means to list of supported
+        notification mechanisms in etcd in '/notification_medium' directory.
+    *** The alerting module will expose apis for managing configurations and
+        subscriptions for each of the notification medium which will be proxied
+        to by tendrl-api.
+        The way the api works is as follows:
+        **** The configuration if applicable will link to the tendrl maintained
```

Comment:
I think we can implement this alerting specific way of configuration for now, but eventually, I think it should be tackled via https://github.com/Tendrl/specifications/issues/29.